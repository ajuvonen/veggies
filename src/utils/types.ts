import type {DateTime} from 'luxon';

export type Locale = 'en' | 'fi';

export enum Category {
  Fruit = 'Fruit',
  Vegetable = 'Vegetable',
  Leafy = 'Leafy',
  Root = 'Root',
  Bean = 'Bean',
  Grain = 'Grain',
  Mushroom = 'Mushroom',
}

export type Favorites = {
  [key in Category]: readonly [string, number][];
};

export type Week = {
  veggies: string[];
  readonly startDate: DateTime;
};

export type Challenge = {
  readonly startDate: DateTime;
  veggie: string | undefined;
};

export type Settings = {
  allergens: string[];
  locale: Locale;
  showChartAnimations: boolean;
  suggestionCount: number;
  summaryViewedDate: DateTime | null;
};

export type Listing = {
  readonly veggie: string;
  readonly category: Category;
};

export type TranslatedListing = Listing & {
  readonly translation: string;
  readonly synonyms: string[];
};

export enum AchievementLevel {
  NoAchievement,
  Bronze,
  Silver,
  Gold,
  Platinum,
}

export type Achievements = {
  allOnRed: AchievementLevel;
  botanicalBerries: AchievementLevel;
  challengeAccepted: AchievementLevel;
  committed: AchievementLevel;
  completionist: AchievementLevel;
  experimenterBean: AchievementLevel;
  experimenterFruit: AchievementLevel;
  experimenterGrain: AchievementLevel;
  experimenterLeafy: AchievementLevel;
  experimenterMushroom: AchievementLevel;
  experimenterRoot: AchievementLevel;
  experimenterVegetable: AchievementLevel;
  goNuts: AchievementLevel;
  hotStreak: AchievementLevel;
  lemons: AchievementLevel;
  thirtyVeggies: AchievementLevel;
  thousandsOdd: AchievementLevel;
  thousandsEven: AchievementLevel;
};

export type WeekData = {
  atMostVeggies: number;
  challenge: string | undefined;
  firstTimeVeggies: string[];
  firstWeek: boolean;
  hotStreak: number;
  mean: number;
  previousWeekCount: number;
  veggies: string[];
  weekNumber: string;
};

export type SummaryItem = {
  emoji: string;
  translationKey: string;
  translationParameters: (string | number)[];
};
