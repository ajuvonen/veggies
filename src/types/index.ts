export type WeekData = {
  veggies: string[];
  weekNumber: string;
};

export type SummaryItem = {
  emoji: string;
  translationKey: string;
  translationParameters: (string | number)[];
};
