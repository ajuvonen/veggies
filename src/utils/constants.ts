import type {InjectionKey, Ref} from 'vue';
import type {UseMemoizeReturn} from '@vueuse/core';
import {Category, type Locale} from './types';

export const APP_URL = 'https://eatyourveggies.app';

export const BLUESKY_URL = 'https://bsky.app/profile/eatyourveggies.app';

export const PLAY_STORE_URL =
  'https://play.google.com/store/apps/details?id=app.eatyourveggies.twa';

export const BUILD_TIME = __VITE_BUILD_TIME__;

export const DEFAULT_LOCALE: Locale = 'en';

export const DEFAULT_SETTINGS = {
  allergens: [] as string[],
  locale: DEFAULT_LOCALE,
  showChartAnimations: true,
  suggestionCount: 10,
} as const;

export const KEYS = {
  challenge: Symbol() as InjectionKey<Ref<string | undefined>>,
  dropdownStyles: Symbol() as InjectionKey<
    UseMemoizeReturn<string, [active: boolean, selected: boolean]>
  >,
} as const;

export const LOCALES: readonly Locale[] = ['en', 'fi'] as const;

export const COLORS = {
  chartColors: ['#f0f9ff', '#bae6fd', '#38bdf8', '#0284c7', '#075985', '#082f49', '#0B1623'],
  chartColorsAlternate: [
    '#fdc5f5',
    '#f7aef8',
    '#b388eb',
    '#8093f1',
    '#79b8f4',
    '#72ddf7',
    '#bae6fd',
  ],
  offWhite: '#f9fafb',
  darkBlue: '#0c4a6e',
} as const;

export const CATEGORY_EMOJI = {
  [Category.Fruit]: 'üçé',
  [Category.Vegetable]: 'ü•¶',
  [Category.Leafy]: 'ü•¨',
  [Category.Root]: 'ü•ï',
  [Category.Bean]: 'ü´õ',
  [Category.Grain]: 'üåæ',
  [Category.Mushroom]: 'üçÑ‚Äçüü´',
} as const;

export const FRUITS: readonly string[] = [
  'apple',
  'apricot',
  'asian pear',
  'banana',
  'blackberry',
  'blackcurrant',
  'blood grapefruit',
  'blood orange',
  'blueberry',
  'boysenberry',
  'calamansi',
  'cantaloupe',
  'cape gooseberry',
  'caper',
  'cherimoya',
  'cherry',
  'clementine',
  'cloudberry',
  'coconut',
  'cranberry',
  'date',
  'dragon fruit',
  'durian',
  'fig',
  'finger lime',
  'gooseberry',
  'grape',
  'grapefruit',
  'grenadilla',
  'guava',
  'honeydew melon',
  'jackfruit',
  'kiwi',
  'kumquat',
  'lemon',
  'lime',
  'lingonberry',
  'longan',
  'lychee',
  'mandarin',
  'mango',
  'mangosteen',
  'nectarine',
  'orange',
  'papaya',
  'passion fruit',
  'peach',
  'pear',
  'persimmon',
  'piel de sapo',
  'pineapple',
  'plum',
  'pomegranate',
  'pomelo',
  'prickly pear',
  'quince',
  'rambutan',
  'raspberry',
  'redcurrant',
  'sea buckthorn',
  'starfruit',
  'strawberry',
  'watermelon',
];

export const LEAFIES: readonly string[] = [
  'arugula',
  'basil',
  'bok choy',
  'cabbage',
  'chard',
  'choy sum',
  'cilantro',
  'collard greens',
  'dandelion greens',
  'dill',
  'endive',
  'frisee',
  'holy basil',
  'iceberg lettuce',
  'kale',
  'lettuce',
  'little gem lettuce',
  'mache',
  'mint',
  'mizuna',
  'mustard greens',
  'napa cabbage',
  'oak leaf lettuce',
  'parsley',
  'pea sprouts',
  'radicchio',
  'red cabbage',
  'red leaf lettuce',
  'romaine',
  'savoy cabbage',
  'sorrel',
  'spinach',
  'sprouts',
  'thai basil',
  'thyme',
  'water spinach',
  'watercress',
];

export const ROOTS: readonly string[] = [
  'bamboo shoots',
  'beet',
  'black radish',
  'black salsify',
  'carrot',
  'celeriac',
  'chives',
  'daikon',
  'fennel',
  'garlic',
  'ginger',
  'horseradish',
  'jerusalem artichoke',
  'jicama',
  'kohlrabi',
  'leek',
  'manioc',
  'onion',
  'parsley root',
  'parsnip',
  'potato',
  'radish',
  'red onion',
  'rutabaga',
  'scallion',
  'shallot',
  'sweet potato',
  'taro',
  'turnip',
  'wasabi',
  'white onion',
  'yam',
  'yellow beet',
];

export const VEGETABLES: readonly string[] = [
  'artichoke',
  'asparagus',
  'avocado',
  'bell pepper',
  'bitter melon',
  'broccoli',
  'broccolini',
  'brussels sprouts',
  'butternut squash',
  'cauliflower',
  'celery',
  'chayote',
  'cherry tomato',
  'chili pepper',
  'corn',
  'cucumber',
  'eggplant',
  'hokkaido pumpkin',
  'jalapeno',
  'nori',
  'okra',
  'olive',
  'plantain',
  'pointed pepper',
  'pumpkin',
  'rhubarb',
  'spaghetti squash',
  'tomatillo',
  'tomato',
  'wakame',
  'wax gourd',
  'zucchini',
];

export const MUSHROOMS: readonly string[] = [
  'black trumpet mushroom',
  'champignon',
  'chanterelle',
  'enoki',
  'hedgehog mushroom',
  'king oyster mushroom',
  'matsutake',
  'morel',
  'oyster mushroom',
  'porcini',
  'portobello',
  'shiitake',
  'shimeji',
  'wild mushroom',
  'wood ear mushroom',
  'yellowfoot mushroom',
];

export const BEANS: readonly string[] = [
  'black bean',
  'black lentil',
  'black-eyed pea',
  'brown lentil',
  'butter bean',
  'cannellini bean',
  'chickpea',
  'cranberry bean',
  'edamame',
  'fava bean',
  'flageolet bean',
  'green bean',
  'green lentil',
  'kidney bean',
  'mung bean',
  'pea',
  'pinto bean',
  'red bean',
  'red lentil',
  'scarlet runner bean',
  'snow pea',
  'sugar snap pea',
  'soybean',
  'white bean',
  'yardlong bean',
  'yellow lentil',
];

export const GRAINS: readonly string[] = [
  'almond',
  'amaranth',
  'barley',
  'brazil nut',
  'buckwheat',
  'bulgur',
  'cashew nut',
  'chestnut',
  'chia seed',
  'couscous',
  'farro',
  'flaxseed',
  'hazelnut',
  'hemp seed',
  'kamut',
  'macadamia nut',
  'millet',
  'oat',
  'peanut',
  'pecan nut',
  'pine nut',
  'pistachio nut',
  'poppy seed',
  'pumpkin seed',
  'quinoa',
  'rice',
  'rye',
  'safflower seed',
  'sesame seed',
  'sorghum',
  'spelt',
  'sunflower seed',
  'teff',
  'triticale',
  'walnut',
  'wheat',
  'wild rice',
];

export const ALL_VEGGIES: readonly string[] = [
  ...FRUITS,
  ...VEGETABLES,
  ...LEAFIES,
  ...ROOTS,
  ...BEANS,
  ...GRAINS,
  ...MUSHROOMS,
];

export const CITRUSES: readonly string[] = [
  'blood grapefruit',
  'blood orange',
  'calamansi',
  'clementine',
  'finger lime',
  'grapefruit',
  'kumquat',
  'lemon',
  'lime',
  'mandarin',
  'orange',
  'pomelo',
];

export const RED_VEGGIES: readonly string[] = [
  'apple',
  'beet',
  'bell pepper',
  'blood grapefruit',
  'blood orange',
  'cherry tomato',
  'cherry',
  'chili pepper',
  'cranberry',
  'lingonberry',
  'pointed pepper',
  'pomegranate',
  'radish',
  'raspberry',
  'red bean',
  'red cabbage',
  'red onion',
  'redcurrant',
  'strawberry',
  'tomato',
  'watermelon',
];

export const NUTS: readonly string[] = [
  'almond',
  'brazil nut',
  'cashew nut',
  'chestnut',
  'coconut',
  'hazelnut',
  'macadamia nut',
  'peanut',
  'pecan nut',
  'pine nut',
  'pistachio nut',
  'walnut',
];

export const BOTANICAL_BERRIES: readonly string[] = [
  'banana',
  'bell pepper',
  'blood grapefruit',
  'blood orange',
  'blueberry',
  'cantaloupe',
  'cape gooseberry',
  'cherry tomato',
  'chili pepper',
  'clementine',
  'cranberry',
  'cucumber',
  'eggplant',
  'grape',
  'grapefruit',
  'guava',
  'honeydew melon',
  'kiwi',
  'kumquat',
  'lemon',
  'lime',
  'mandarin',
  'orange',
  'papaya',
  'passion fruit',
  'piel de sapo',
  'plantain',
  'pointed pepper',
  'pomelo',
  'pumpkin',
  'tomatillo',
  'tomato',
  'watermelon',
  'zucchini',
];
