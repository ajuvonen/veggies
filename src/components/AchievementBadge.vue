<script setup lang="ts">
import {CATEGORY_EMOJI} from '@/utils/constants';
import {AchievementLevel, type Achievements} from '@/utils/types';

defineProps<{
  achievement: keyof Achievements;
  level: AchievementLevel;
  active: boolean;
}>();

type BadgeProps = Record<
  keyof Achievements,
  Partial<
    Record<
      AchievementLevel,
      {
        textProps: (string | number)[];
        emoji: string;
      }
    >
  >
>;

const badgeProps: BadgeProps = {
  challengeAccepted: {
    [AchievementLevel.Bronze]: {
      textProps: [5],
      emoji: 'üöÅ',
    },
    [AchievementLevel.Silver]: {
      textProps: [10],
      emoji: '‚úàÔ∏è',
    },
    [AchievementLevel.Gold]: {
      textProps: [20],
      emoji: 'üöÄ',
    },
  },
  committed: {
    [AchievementLevel.Bronze]: {
      textProps: [3],
      emoji: 'üå±',
    },
    [AchievementLevel.Silver]: {
      textProps: [6],
      emoji: 'ü™¥',
    },
    [AchievementLevel.Gold]: {
      textProps: [12],
      emoji: 'üéÑ',
    },
  },
  completionist: {
    [AchievementLevel.Bronze]: {
      textProps: [40],
      emoji: 'üê£',
    },
    [AchievementLevel.Silver]: {
      textProps: [80],
      emoji: 'üêß',
    },
    [AchievementLevel.Gold]: {
      textProps: [150],
      emoji: 'ü¶Ö',
    },
  },
  experimenterBean: {
    [AchievementLevel.Gold]: {
      textProps: [15],
      emoji: CATEGORY_EMOJI.Bean,
    },
  },
  experimenterFruit: {
    [AchievementLevel.Gold]: {
      textProps: [15],
      emoji: CATEGORY_EMOJI.Fruit,
    },
  },
  experimenterGrain: {
    [AchievementLevel.Gold]: {
      textProps: [15],
      emoji: CATEGORY_EMOJI.Grain,
    },
  },
  experimenterLeafy: {
    [AchievementLevel.Gold]: {
      textProps: [15],
      emoji: CATEGORY_EMOJI.Leafy,
    },
  },
  experimenterRoot: {
    [AchievementLevel.Gold]: {
      textProps: [15],
      emoji: CATEGORY_EMOJI.Root,
    },
  },
  experimenterVegetable: {
    [AchievementLevel.Gold]: {
      textProps: [15],
      emoji: CATEGORY_EMOJI.Vegetable,
    },
  },
  favorite: {
    [AchievementLevel.Bronze]: {
      textProps: [5],
      emoji: 'üò∫',
    },
    [AchievementLevel.Silver]: {
      textProps: [20],
      emoji: 'üòΩ',
    },
    [AchievementLevel.Gold]: {
      textProps: [50],
      emoji: 'üòª',
    },
  },
  hotStreak: {
    [AchievementLevel.Bronze]: {
      textProps: [5],
      emoji: 'üå∂Ô∏è',
    },
    [AchievementLevel.Silver]: {
      textProps: [10],
      emoji: 'üî•',
    },
    [AchievementLevel.Gold]: {
      textProps: [20],
      emoji: 'üåã',
    },
  },
  thirtyVeggies: {
    [AchievementLevel.Gold]: {
      textProps: [30],
      emoji: 'üéâ',
    },
    [AchievementLevel.Platinum]: {
      textProps: [40],
      emoji: 'üíé',
    },
  },
};
</script>
<template>
  <div
    :aria-disabled="!active"
    v-tippy="{
      aria: null,
      content: $t(
        `achievements.${achievement}.ariaLabel`,
        badgeProps[achievement][level]!.textProps,
      ),
    }"
    :aria-label="
      $t(`achievements.${achievement}.ariaLabel`, badgeProps[achievement][level]!.textProps)
    "
    :data-test-id="`badge-${achievement}-${level}`"
    class="badge"
    role="img"
  >
    <div
      class="badge__background"
      :class="[`badge__background--${AchievementLevel[level]}`]"
      aria-hidden="true"
    >
      <div class="badge__emoji">
        {{ badgeProps[achievement][level]!.emoji }}
      </div>
    </div>
    <div aria-hidden="true" class="badge__text">
      {{ $t(`achievements.${achievement}.badgeText`, badgeProps[achievement][level]!.textProps) }}
    </div>
  </div>
</template>
<style scoped>
.badge {
  @apply relative select-none aspect-square self-center max-w-40;
  filter: drop-shadow(0px 0px 3px rgba(0, 0, 0, 0.3));
  flex: 0 0 calc(33% - 5px);

  &[aria-disabled='true'] {
    opacity: 0.5;
  }
}

.badge__background {
  @apply relative w-full h-full rounded-full border-4 overflow-hidden text-[17cqmin] sm:text-[14cqmin];
  @apply flex items-center justify-center;
  box-shadow: inset 0 0 15px rgba(0, 0, 0, 0.3);
  text-shadow: 1px 1px 1px #334155;
}

.badge__emoji {
  mix-blend-mode: luminosity;
}

.badge__text {
  @apply absolute bottom-[3cqmin] min-w-full text-nowrap rounded-md text-center text-xs uppercase;
  @apply bg-sky-950 text-slate-50;
}

.badge__background--Bronze {
  background: linear-gradient(135deg, #cd7f32, #b87333);
  border-color: #d48b47;
}

.badge__background--Silver {
  background: linear-gradient(135deg, #c0c0c0, #a9a9a9);
  border-color: #c9c9c9;
}

.badge__background--Gold {
  background: linear-gradient(135deg, #d4af37, #b89b36, #8c7d33);
  border-color: #d9b43b;
}

.badge__background--Platinum {
  background: linear-gradient(135deg, #ffffff, #d9d9d9);
  border-color: #f0f0f0;

  .badge__emoji {
    mix-blend-mode: normal;
  }
}

.badge__background::after {
  content: '';
  @apply absolute top-0 left-0 w-full h-full rounded-full;
  background: radial-gradient(circle at top left, rgba(255, 255, 255, 0.5), transparent);
  mask-image: radial-gradient(circle, white 30%, transparent 60%);
  mask-size: 150%;
  mask-position: 30% 30%;
}

.badge__background::before {
  content: '';
  @apply absolute w-full h-full rounded-full border-2 border-dotted border-inherit;
}
</style>
