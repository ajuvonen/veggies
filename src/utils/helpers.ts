import {useMemoize} from '@vueuse/core';
import type {ChartOptions, ChartType, Scale} from 'chart.js';
import type {Context} from 'chartjs-plugin-datalabels';
import {DateTime} from 'luxon';
import {mergeDeep, sample} from 'remeda';
import {
  ALL_VEGGIES,
  BEANS,
  CATEGORY_EMOJI,
  COLORS,
  FRUITS,
  GRAINS,
  LEAFIES,
  MUSHROOMS,
  ROOTS,
  VEGETABLES,
} from '@/utils/constants';
import {AchievementLevel, Category} from '@/utils/types';

export const getCategoryForVeggie = useMemoize((veggie: string) => {
  if (FRUITS.includes(veggie)) {
    return Category.Fruit;
  } else if (VEGETABLES.includes(veggie)) {
    return Category.Vegetable;
  } else if (LEAFIES.includes(veggie)) {
    return Category.Leafy;
  } else if (ROOTS.includes(veggie)) {
    return Category.Root;
  } else if (BEANS.includes(veggie)) {
    return Category.Bean;
  } else if (GRAINS.includes(veggie)) {
    return Category.Grain;
  } else if (MUSHROOMS.includes(veggie)) {
    return Category.Mushroom;
  }
  return undefined;
});

export const getChartOptions = <T extends ChartType>(
  grids: boolean,
  stacked: boolean,
  withIcons: boolean,
  showChartAnimations: boolean,
  overrides: Partial<ChartOptions<T>> = {},
) =>
  mergeDeep(
    {
      animation: !showChartAnimations ? false : undefined,
      responsive: true,
      maintainAspectRatio: !grids,
      normalized: true,
      layout: {
        padding: 0,
      },
      scales: grids
        ? {
            y: {
              beginAtZero: true,
              ticks: {
                precision: 0,
                color: COLORS.offWhite,
              },
              stacked,
            },
            y1: {
              position: 'right',
              beginAtZero: true,
              ticks: {
                precision: 0,
                color: COLORS.offWhite,
              },
              stacked,
              afterBuildTicks: (axis: Scale) => {
                axis.ticks = [...axis.chart.scales.y.ticks];
                axis.min = axis.chart.scales.y.min;
                axis.max = axis.chart.scales.y.max;
              },
            },
            x: {
              beginAtZero: true,
              ticks: {
                precision: 0,
                color: COLORS.offWhite,
              },
              stacked,
            },
          }
        : undefined,
      plugins: {
        title: {
          display: false,
        },
        legend: {
          display: false,
        },
        datalabels: {
          ...(withIcons
            ? {
                anchor: 'center',
                align: 'center',
                font: {
                  size: 25,
                },
                textShadowColor: '#fff',
                textShadowBlur: 5,
                // eslint-disable-next-line @typescript-eslint/no-explicit-any
                formatter: (value: any, {dataset: {label}}: Context) =>
                  value ? CATEGORY_EMOJI[label as Category] : '',
              }
            : {
                display: false,
              }),
        },
        tooltip: {
          animation: showChartAnimations,
          padding: 8,
          titleFont: {
            size: 14,
            weight: 'normal',
          },
          bodyFont: {
            size: 14,
          },
          footerFont: {
            size: 14,
            weight: 'normal',
          },
          displayColors: false,
          backgroundColor: COLORS.darkBlue,
          bodyColor: COLORS.offWhite,
          titleColor: COLORS.offWhite,
        },
      },
    },
    overrides,
  ) as ChartOptions<T>;

export const getRandomVeggie = () => ALL_VEGGIES[Math.floor(Math.random() * ALL_VEGGIES.length)];

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const dateParser = (key: string, value: any) => {
  if (key === 'startDate' && value) {
    return DateTime.fromISO(value.split('T')[0]);
  }
  return value;
};

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const dateReplacer = (key: string, value: any) => {
  if (key === 'startDate' && value) {
    return value.split('T')[0];
  }
  return value;
};

const veggieEmojis: readonly string[] = [
  '🥝',
  '🥥',
  '🍇',
  '🍈',
  '🍉',
  '🍊',
  '🍋',
  '🍌',
  '🍍',
  '🥭',
  '🍎',
  '🍏',
  '🍐',
  '🍑',
  '🍒',
  '🍓',
  '🫐',
  '🍅',
  '🍆',
  '🌽',
  '🌶️',
  '🫑',
  '🥑',
  '🥒',
  '🥬',
  '🥦',
  '🥔',
  '🧄',
  '🧅',
  '🥕',
  '🫛',
  '🥜',
  '🫘',
  '🌿',
  '🍄‍🟫',
  '🌾',
  '🌰',
];
export const getRandomEmojis = (amount: number = 1) => sample(veggieEmojis, amount);

export const achievementLevelHelper = (levels: [number, AchievementLevel][], value: number) =>
  levels.find(([threshold]) => value >= threshold)?.[1] ?? AchievementLevel.NoAchievement;
