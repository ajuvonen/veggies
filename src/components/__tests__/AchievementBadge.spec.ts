import {describe, it, expect} from 'vitest';
import {mount} from '@vue/test-utils';
import AchievementBadge from '@/components/AchievementBadge.vue';
import {AchievementLevel} from '@/utils/types';

describe('AchievementBadge', () => {
  it('renders active', () => {
    const wrapper = mount(AchievementBadge, {
      props: {
        achievement: 'completionist',
        level: AchievementLevel.Gold,
        active: true,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders inactive', () => {
    const wrapper = mount(AchievementBadge, {
      props: {
        achievement: 'hotStreak',
        level: AchievementLevel.Silver,
        active: false,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('does not render gradient when achievement is active', () => {
    const wrapper = mount(AchievementBadge, {
      props: {
        achievement: 'goNuts',
        level: AchievementLevel.Gold,
        active: true,
        degree: 360,
      },
    });
    expect(wrapper.find('.badge__overlay').exists()).toBe(false);
  });

  it('renders gradient', async () => {
    const wrapper = mount(AchievementBadge, {
      props: {
        achievement: 'thirtyVeggies',
        level: AchievementLevel.Gold,
        active: false,
        degree: 180,
      },
    });

    expect(wrapper.find('.badge__overlay').attributes('style')).toBe(
      'mask-image: conic-gradient(transparent 0deg 180deg, black 180deg 360deg);',
    );
    await wrapper.setProps({degree: 120});
    expect(wrapper.find('.badge__overlay').attributes('style')).toBe(
      'mask-image: conic-gradient(transparent 0deg 120deg, black 120deg 360deg);',
    );
  });
});
