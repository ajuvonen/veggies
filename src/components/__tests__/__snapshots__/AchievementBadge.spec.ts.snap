// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AchievementBadge > renders active 1`] = `
"<li data-v-06381b3c="" aria-disabled="false" aria-label="150 unique veggies" data-test-id="badge-completionist-3" class="badge__completionist badge" role="img">
  <div data-v-06381b3c="" class="badge__background badge__background--Gold" aria-hidden="true">
    <div data-v-06381b3c="" class="badge__emoji">ğŸ¦…</div>
  </div>
  <!--v-if-->
  <div data-v-06381b3c="" aria-hidden="true" class="badge__text">150 veggies</div>
</li>"
`;

exports[`AchievementBadge > renders inactive 1`] = `
"<li data-v-06381b3c="" aria-disabled="true" aria-label="10 week streak of over 30 veggies" data-test-id="badge-hotStreak-2" class="badge__hotStreak badge" role="img">
  <div data-v-06381b3c="" class="badge__background badge__background--Silver" aria-hidden="true">
    <div data-v-06381b3c="" class="badge__emoji">ğŸ”¥</div>
  </div>
  <div data-v-06381b3c="" class="badge__overlay" style="mask-image: conic-gradient(transparent 0deg 0deg, black 0deg 360deg);"></div>
  <div data-v-06381b3c="" aria-hidden="true" class="badge__text">10 Weeks</div>
</li>"
`;
